forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt

```diff
@@ -32,84 +30,60 @@ internal object LDContextKind {
-        // shouldn't be used directly.
-        set(value) {
-            if (value == null) {
-                internalLogger.w("[LaunchDarkly] vaultType is being reset to null. This should only happen while unit testing!")
-                internalVaultType = null
-            } else if (internalVaultType == null) {
-                internalVaultType = value
-            } else {
-                throw Error("vaultType can only be set once!")
-            }
-        }
-
-    internal fun initialize(app: Application, logger: Log, dataSource: TestData? = null) {
-        val ldConfig = if (dataSource != null) {
-            LDConfig.Builder()
-                .mobileKey(LD_MOBILE_KEY)
-                .dataSource(dataSource)
-                .build()
-        } else {
-            LDConfig.Builder()
-                .mobileKey(LD_MOBILE_KEY)
-                .build()
-        }
+    internal fun initialize(app: Application, ldConfig: LDConfig) {
         val contextKind = ContextKind.of(LDContextKind.SERVICE)
         val context = LDContext.create(contextKind, LDContexts.ANDROID_CONTEXT)
         client = LDClient.init(app, ldConfig, context, 0)
-
-        internalLogger = logger
     }
 
-    // vaultType is instantiated lazily and is a singleton. Once we set the vault type once, we don't
-    // want to overwrite it! We must take in the application as a parameter, which means that a
-    // ForagePINEditText must be rendered before any of the ForageSDKApi functions are called.
-    internal fun getVaultProvider(): VaultType {
-        if (vaultType != null) {
-            return vaultType as VaultType
-        }
-        // default to 100% VGS usage in case LD flag retrieval fails
-        val defaultVal = 0.0
-        val vaultPercent =
-            client?.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, defaultVal) ?: defaultVal
-        internalLogger.i("[LaunchDarkly] Vault percent of $vaultPercent return from LD")
-        val randomNum = Math.random() * 100
-
-        vaultType = if (randomNum < vaultPercent) {
-            VaultType.BT_VAULT_TYPE
-        } else {
-            VaultType.VGS_VAULT_TYPE
-        }
-        internalLogger.i("[LaunchDarkly] Vault type set to $vaultType")
-        return vaultType as VaultType
+    internal fun getVaultProvider(logger: Log = Log.getSilentInstance()): VaultType {
+        val vaultPercent = client?.doubleVariation(
+            LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG,
+            ALWAYS_VGS_PERCENT
+        ) ?: ALWAYS_VGS_PERCENT
+        logger.i("[LaunchDarkly] Vault percent of $vaultPercent return from LD")
+
+        // convert the flag percent into an answer to which vault provider to use
+        val vaultType = computeVaultType(vaultPercent)
+        logger.i("[LaunchDarkly] Vault type set to $vaultType")
+
+        // return vault provider derived from the novel or cached flag value
```

Comment:
We don't cache that value anymore right?

Reply:
fixed in [a76ec00](https://github.com/teamforage/forage-android-sdk/pull/128/commits/a76ec003a8aaf3e2d8f8e10856294dc92e19262a)